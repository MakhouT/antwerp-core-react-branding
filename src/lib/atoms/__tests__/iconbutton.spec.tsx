import * as React from "react";

import * as A from "../..";
import { renderAndMatchSnapshot } from "../../../__tests__/__fixtures__/testUtils";

describe("Button", () => {
    it("Renders default icon button", () => {
        // Act
        const element = (
            <A.IconButton
                icon="check"
                text="testbutton"
            />);

        // Act and assert
        const tree = renderAndMatchSnapshot(element);

        expect(tree.props.className).toContain("has-icon");
    });

    it("Renders default icon button, icon on the left when location not specified", () => {
        // Act
        const element = (
            <A.IconButton
                icon="check"
                text="testbutton"
            />);

        // Act and assert
        const tree = renderAndMatchSnapshot(element);

        expect(tree.props.className).toContain("has-icon-left");
    });

    it("Renders default icon button, icon on the right", () => {
        // Act
        const element = (
            <A.IconButton
                icon="check"
                text="testbutton"
                location={A.Location.Right}
            />);

        // Act and assert
        const tree = renderAndMatchSnapshot(element);

        expect(tree.props.className).toContain("has-icon-right");
    });

    it("Renders default icon button, icon on the right", () => {
        // Act
        const element = (
            <A.IconButton
                icon="check"
                location={A.Location.Right}
            />);

        // Act and assert
        const tree = renderAndMatchSnapshot(element);

        expect(tree.props.className).toContain("has-icon");
        expect(tree.props.className).not.toContain("has-icon-left");
    });
});
