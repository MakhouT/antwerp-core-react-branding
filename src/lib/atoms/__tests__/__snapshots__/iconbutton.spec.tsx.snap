// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Renders default icon button 1`] = `
<button
  autoFocus={undefined}
  className="a-button has-icon-left"
  disabled={undefined}
  onClick={[Function]}
  style={undefined}
  tabIndex={undefined}
  type="button"
>
  <i
    className="fa fa-check"
    onClick={[Function]}
    style={undefined}
  />
  testbutton
</button>
`;

exports[`Button Renders default icon button, icon on the left when location not specified 1`] = `
<button
  autoFocus={undefined}
  className="a-button has-icon-left"
  disabled={undefined}
  onClick={[Function]}
  style={undefined}
  tabIndex={undefined}
  type="button"
>
  <i
    className="fa fa-check"
    onClick={[Function]}
    style={undefined}
  />
  testbutton
</button>
`;

exports[`Button Renders default icon button, icon on the right 1`] = `
<button
  autoFocus={undefined}
  className="a-button has-icon-right"
  disabled={undefined}
  onClick={[Function]}
  style={undefined}
  tabIndex={undefined}
  type="button"
>
  <i
    className="fa fa-check"
    onClick={[Function]}
    style={undefined}
  />
  testbutton
</button>
`;

exports[`Button Renders default icon button, icon on the right 2`] = `
<button
  autoFocus={undefined}
  className="a-button has-icon-right"
  disabled={undefined}
  onClick={[Function]}
  style={undefined}
  tabIndex={undefined}
  type="button"
>
  <i
    className="fa fa-check"
    onClick={[Function]}
    style={undefined}
  />
</button>
`;
